gamemodes/terrortown/gamemode/server/sv_armor.lua

```diff
@@ -0,0 +1,146 @@
+local plymeta = FindMetaTable("Player")
+if not plymeta then
+	Error("FAILED TO FIND PLAYER TABLE")
+
+	return
+end
+
+ARMOR = {}
+
+
+-- SET UP NETWORK STRINGS
+util.AddNetworkString("ttt2_sync_armor")
+util.AddNetworkString("ttt2_sync_armor_max")
+
+
+-- SET UP CONVARS
+ARMOR.cv = {}
+ARMOR.cv.armor_on_spawn = CreateConVar('ttt_armor_on_spawn', 0, {FCVAR_NOTIFY, FCVAR_ARCHIVE})
+ARMOR.cv.armor_is_reinforced_enabled = CreateConVar('ttt_armor_is_reinforced_enabled', 1, {FCVAR_NOTIFY, FCVAR_ARCHIVE})
+ARMOR.cv.armor_threshold_for_reinforced = CreateConVar('ttt_armor_threshold_for_reinforced', 50, {FCVAR_NOTIFY, FCVAR_ARCHIVE})
+ARMOR.cv.armor_damage_block_pct = CreateConVar('ttt_armor_damage_block_pct', 0.2, {FCVAR_NOTIFY, FCVAR_ARCHIVE})
+ARMOR.cv.armor_damage_health_pct = CreateConVar('ttt_armor_damage_health_pct', 0.7, {FCVAR_NOTIFY, FCVAR_ARCHIVE})
+ARMOR.cv.armor_classic = CreateConVar('ttt_armor_classic', 0, {FCVAR_NOTIFY, FCVAR_ARCHIVE})
+ARMOR.cv.item_armor_value = CreateConVar('ttt_item_armor_value', 30, {FCVAR_NOTIFY, FCVAR_ARCHIVE})
+
+cvars.AddChangeCallback("ttt_armor_classic", function(cv, old, new)
+    SetGlobalBool("ttt_armor_classic", tobool(tonumber(new)))
+end)
+cvars.AddChangeCallback("ttt_armor_is_reinforced_enabled", function(cv, old, new)
+    SetGlobalBool("ttt_armor_is_reinforced_enabled", tobool(tonumber(new)))
+end)
+cvars.AddChangeCallback("ttt_armor_threshold_for_reinforced", function(cv, old, new)
+    SetGlobalInt("ttt_armor_threshold_for_reinforced", tonumber(new))
+end)
+
+
+-- SERVERSIDE ARMOR FUNCTIONS
+---
+-- Sets the armor to a specific value
+-- @param number armor the new armor to be set
+-- @realm server
+function plymeta:SetArmor(armor)
+    self.armor = armor
+    self.armor_is_reinforced = ARMOR.cv.armor_is_reinforced_enabled:GetBool() and self:Armor() > ARMOR.cv.armor_threshold_for_reinforced:GetInt()
+
+    net.Start("ttt2_sync_armor")
+    net.WriteUInt(math.Round(self.armor), 16)
+    net.WriteBool(self.armor_is_reinforced)
+    net.Send(self)
+end
+
+---
+-- Sets the max armor to a specific value
+-- @param number armor_max the new max armor to be set
+-- @realm server
+function plymeta:SetMaxArmor(armor_max)
```

Comment:
This will not change the current armor accordingly

Reply:
MaxArmor is only used for HUDs